### PR TITLE
solution

### PR DESCRIPTION
**Pull request by VladyslavOladko**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_ls = []
+    for person in people:
+        person_ls.append(Person(person["name"], person["age"]))
+
+    for i in people:
```
Comment (Vasyl-Smutok) [inline app/main.py:15]: `і` bad variable name

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_ls = []
+    for person in people:
+        person_ls.append(Person(person["name"], person["age"]))
```
Comment (Vasyl-Smutok) [inline app/main.py:13]: I recommend using here `list comprehension`

**Review by Vasyl-Smutok — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,20 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
```
Comment (ArsenPidhoretskyi) [inline app/main.py:2]: provide typing
people: dict[str, Person]

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,20 @@
...
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_ls = [Person(person["name"], person["age"]) for person in people]
+
+    for person in people:
+        if "wife" in person and person["wife"] is not None:
```
Comment (ArsenPidhoretskyi) [inline app/main.py:14]: You can simplify the conditions by using the .get method on the dict
The same with husband

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**